spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -82,6 +82,68 @@ The xref:getting-started#getting-started[Getting Started] section shows you how
 
+=== Stream Application DSL
+
+The Stream Pipeline DSL described in the previous section automatically sets the input and output binding properties of each Spring Cloud Stream application.
+This can be done because there is only one input and/or output destination in a Spring Cloud Stream application that uses the provided binding interface of a `Source`, `Processor`, or `Sink`.
+However, a Spring Cloud Stream application can define a custom binding interface such as the one shown below
+
+[source,java]
+----
+public interface Barista {
+
+    @Input
+    SubscribableChannel orders();
+
+    @Output
+    MessageChannel hotDrinks();
+
+    @Output
+    MessageChannel coldDrinks();
+}
+----
+
+or as is common when creating a Kafka Streams application,
+
+[source,java]
+----
+interface KStreamKTableBinding {
+
+    @Input
+    KStream<?, ?> inputStream();
+
+    @Input
+    KTable<?, ?> inputTable();
+}
+----
+
+In these cases with multiple input and output bindings, Data Flow can not make any assumptions about the flow of data from one application to another.
+Therefore the developer needs to set the binding properties to 'wire up' the application.
+The *Stream Application DSL* uses a `comma`, instead of the `pipe symbol`, to indicate that Data Flow should not configure the binding properties of the application.
+For example:
+
+`dataflow:> stream create --definition "orderGeneratorApp, baristaApp, hotDrinkDeliveryApp, coldDrinkDeliveryApp" --name myCafeStream`
+
+There are four applications in this stream.
+ The baristaApp has two output destinations, `hotDrinks` and `coldDrinks` intended to be consumed by the `hotDrinkDeliveryApp` and `coldDrinkDeliveryApp` respectively.
+When deploying this stream, you need to set the binding properties so that the `baristaApp` sends hot drink messages to the `hotDrinkDeliveryApp` and cold drink mesage to the `coldDrinkDeliveryApp`.
+For example
+
+```
+app.baristaApp.spring.cloud.stream.bindings.hotDrinks.destination=hotDrinks
+app.baristaApp.spring.cloud.stream.bindings.coldDrinks.destination=coldDrinks
+app.hotDrinkDeliveryApp.spring.cloud.stream.bindings.input.destination=hotDrinks
+app.coldDrinkDeliveryApp..spring.cloud.stream.bindings.input.destination=coldDrinks
+```
+
+If you want to use consumer groups, you will need to set the Spring Cloud Stream application property `spring.cloud.stream.bindings.<channelName>.producer.requiredGroups` and `spring.cloud.stream.bindings.<channelName>.group` on the producer and consumer applications respectively.
+
+Another common use case for the Stream Application DSL is to deploy a http gateway application that sends a synchronous request/reply message to a Kafka or RabbitMQ application.
+In this case both the http gateway application the Kafka or RabbitMQ application can be a Spring Integration application that does not make use of the Spring Cloud Stream library.
+
+It is also possible to deploy just a single application using the Stream application DSL.
```

Comment:
This can be "You can also deploy a single application that by using the Stream application DSL." The reason for the "You can" bit is to make it more active and to get the reader involved (this is all wrapped up with the idea of agency). "using" to "by using" is to accommodate the folks who haven't yet dropped "by" from "by using" (that bit of the language is in flux). It is nominally correct as it is, but it is not optimal.